File: UsuariosAPI/src/main/java/com/clinica/api/personal_service/model/Especialidad.java
package com.clinica.api.personal_service.model;

import jakarta.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "Especialidad")
public class Especialidad {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_especialidad")
    private Long id;

    @Column(name = "nombre", nullable = false, unique = true, length = 100)
    private String nombre;

    @OneToMany(mappedBy = "especialidad", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Empleado> empleados = new ArrayList<>();

    public Especialidad() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public List<Empleado> getEmpleados() {
        return empleados;
    }

    public void setEmpleados(List<Empleado> empleados) {
        this.empleados = empleados;
    }
}

File: personal-service/src/main/kotlin/com/clinica/api/personal_service/model/Especialidad.kt
package com.clinica.api.personal_service.model

import jakarta.persistence.*

@Entity
@Table(name = "especialidad")
data class Especialidad(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_especialidad")
    var id: Long? = null,

    var nombre: String,

    @ManyToOne
    @JoinColumn(name = "id_doctor")
    var doctor: Doctor
)